### ISSUE TITLE
SC2188 false positive with pseudo device /dev/tcp and /dev/udp 

### ISSUE THREAD

**Issue by logand22** [koalaman/shellcheck#2404]
#### For bugs
- Rule Id: SC2188
- My shellcheck version: 0.8.0
- [x] The rule's wiki page does not already cover this (e.g. https://shellcheck.net/wiki/SC2188)
- [x] I tried on https://www.shellcheck.net/ and verified that this is still a problem on the latest commit

#### Here's a snippet or screenshot that shows the problem:

```sh
#!/usr/bin/env bash 
while ! >/dev/tcp/localhost/3008
do
        echo "Waiting for program to start"
        sleep 1
done
```

#### Here's what shellcheck currently says:
```sh
Line 2:
while ! >/dev/tcp/localhost/3008
        ^-- SC2188 (warning): This redirection doesn't have a command. Move to its command (or use 'true' as no-op).
```

#### Here's what I wanted or expected to see:
No issues detected!

It's my understanding that Shellcheck doesn't understand the pseudo device /dev/tcp and /dev/udp in bash.

**Comment by koalaman** [comment]
ShellCheck is just needlessly worried that the missing command means it got lost somehow, as would be the case in this true positive:

    printf '%s\n' "GET / HTTP/1.1" "Host: foo.com" "" 
        > /dev/tcp/localhost/80

ShellCheck's suggestion to quiet it is explicitly using `true` to clarify that "I just want to open and not write", e.g. `until true > /dev/tcp/..`, but this is also one of the now-you-know type warnings that you could perma-disable in `.shellcheckrc` since you're clearly already comfortable with the semantics of redirections without associated commands.
